check_stability.py

```diff
@@ -483,31 +482,69 @@ class LogHandler(reader.LogHandler):
         Subclasses reader.LogHandler.
         """
         def __init__(self):
-            self.results = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+            self.results = []
+
+        def find_or_create_test(self, data):
+            for test in self.results:
```

Comment:
This ends up being quadratic in the number of tests, which is bad. We could use an OrderedDict to store the top-level results and avoid this problem.